Framework/DataHandling/src/SaveDiffCal.cpp

```diff
@@ -87,11 +86,26 @@ std::map<std::string, std::string> SaveDiffCal::validateInputs() {
         result["MaskWorkspace"] = "Must have equal or less number of spectra as the table has rows";
       }
     }
+  } else {
+    GroupingWorkspace_const_sptr groupingWS = getProperty("GroupingWorkspace");
+    MaskWorkspace_const_sptr maskWS = getProperty("MaskWorkspace");
+
+    if ((!groupingWS) && (!maskWS)) {
```

Comment:
What if we have a grouping workspace and do not have a masking workspace?

Reply:
The python script in the description runs with only a GroupingWorkspace and an output filename. It works just fine that way. Since all of the workspaces are optional, this checks that at least one of them is present.